RecoParticleFlow/PFProducer/interface/PFBlockAlgo.h

```diff
@@ -76,10 +69,6 @@ class PFBlockAlgo {
   void packLinks(reco::PFBlock& block, 
 		 const std::unordered_map<std::pair<unsigned int,unsigned int>,double>& links) const; 
   
-  /// Avoid to check links when not useful
-  inline bool linkPrefilter(const reco::PFBlockElement* last, 
```

Comment:
This method is removed because it is never used. In fact, it is the ```BlockElementLinkerBase::linkPrefilter(...)``` that is called instead.

However, if I am not missing anything in the logic and the 
```C++
std::vector<std::unique_ptr<BlockElementLinkerBase>> linkTests_;
````
does not specialize into any of the clasess derived from BlockElementLinkerBase, that ```BlockElementLinkerBase::linkPrefilter(...)``` always identically returns ```true```, see https://cmssdt.cern.ch/lxr/source/RecoParticleFlow/PFProducer/interface/BlockElementLinkerBase.h#0018

Could you please verify? If so, some further cleaning is possible (or some fix, if the original intention was different).

@bendavid @hatakeyamak

Reply:
Hi Andrea, it's not so simple I think. Some of the derived linkers implement this method in a non-trivial way, and thanks to the virtual table these are actually used.

Reply:
> Hi Andrea, it's not so simple I think on first glance. Some of the derived linkers implement this method in a non-trivial way, and thanks to the virtual table these are actually used.

Thank you @guitargeek 
Indeed I was expecting something similar, but it was not straightforward to me to find how those derived linkers enter in the game